vita3k/gui/src/game_selector.cpp

```diff
@@ -65,20 +65,61 @@ bool refresh_game_list(GuiState &gui, HostState &host) {
     return true;
 }
 
-static constexpr auto MENUBAR_HEIGHT = 19;
+inline uint64_t current_time() {
+    return std::chrono::duration_cast<std::chrono::seconds>(
+        std::chrono::high_resolution_clock::now().time_since_epoch())
+        .count();
+}
+
+static std::map<std::string, uint64_t> last_time;
+static const float MENUBAR_HEIGHT = 22.f;
 
 void draw_game_selector(GuiState &gui, HostState &host) {
     const ImVec2 display_size = ImGui::GetIO().DisplaySize;
+    const auto scal = ImVec2(display_size.x / 960.0f, display_size.y / 544.0f);
 
     ImGui::SetNextWindowPos(ImVec2(0, MENUBAR_HEIGHT), ImGuiCond_Always);
     ImGui::SetNextWindowSize(ImVec2(display_size.x, display_size.y - MENUBAR_HEIGHT), ImGuiCond_Always);
-    if (gui.current_background)
+    if (gui.current_background || !gui.theme_backgrounds.empty())
         ImGui::SetNextWindowBgAlpha(host.cfg.background_alpha);
     ImGui::Begin("Game Selector", nullptr, ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoBringToFrontOnFocus | ImGuiWindowFlags_NoSavedSettings);
 
-    if (gui.current_background) {
+    if (!gui.theme_backgrounds["start"].empty()) {
+        if (ImGui::IsAnyWindowHovered() || ImGui::IsAnyItemActive() || ImGui::IsAnyItemHovered())
+            last_time["start"] = 0;
+        else {
+            if (last_time["start"] == 0)
+                last_time["start"] = current_time();
+
+            while (last_time["start"] + host.cfg.theme_start_delay < current_time()) {
+                last_time["start"] += host.cfg.theme_start_delay;
+                gui.theme.start_screen = true;
+            }
+        }
+    }
+
+    if (gui.current_background)
         ImGui::GetBackgroundDrawList()->AddImage(reinterpret_cast<void *>(gui.current_background),
-            ImVec2(0, 0), display_size);
+            ImVec2(0.f, MENUBAR_HEIGHT), display_size);
+
+    if (host.cfg.use_theme_background && !gui.theme_backgrounds["home"].empty()) {
+        if (ImGui::IsWindowHovered()) {
+            if (last_time["home"] == 0)
+                last_time["home"] = current_time();
+
+            while (last_time["home"] + host.cfg.theme_delay_background < current_time()) {
+                last_time["home"] += host.cfg.theme_delay_background;
+                if (gui.theme_current_bg < uint64_t(gui.theme_backgrounds["home"].size() - 1))
+                    ++gui.theme_current_bg;
+                else
+                    gui.theme_current_bg = 0;
+            }
+        }
+        if (gui.current_background != gui.theme_backgrounds["home"][gui.theme_current_bg])
+            gui.current_background = gui.theme_backgrounds["home"][gui.theme_current_bg];
+    } else {
+        if (gui.current_background != gui.user_background[host.cfg.background_image])
```

Comment:
you did it again

Reply:
for this is same on present master, but no worry, and don't care, this pr is here obselet for moment